openstack-upi/openstack-upi.md

```diff
@@ -0,0 +1,281 @@
+---
+title: openstack-upi
+authors:
+  - "@tomassedovic"
+reviewers:
+  - "@mandre"
+  - "@luis5tb"
+  - "@wking"
+  - "@cuppett"
+approvers:
+  - TBD
+creation-date: 2019-09-10
+last-updated: 2019-09-10
+status: implementable
+---
+
+# OpenStack UPI
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+The initial OpenStack support for OpenShift 4 was around the IPI
+workflow. While this is the most convenient approach for the end-users
+of OpenStack, there are situations where it falls short. People who
+want to use OpenStack UPI today have to fall back on
+the [bare metal][baremetal-upi] document or try to translate concepts
+from the AWS one.
+
+[baremetal-upi]: https://github.com/openshift/installer/blob/master/docs/user/metal/install_upi.md
+
+Similar to AWS and GCP, we want to provide the documentation and
+automation (via Ansible) that is specific to OpenStack -- using the
+terminology and tooling that OpenStack people are familiar with and
+highlighting considerations specific to that platform.
+
+## Motivation
+
+In addition to the general use cases supported by UPI (e.g. control
+over the creation of the resources of the cloud provider, custom
+Ignition configuration), there are additional reasons for having an
+OpenStack-specific UPI.
+
+First, unlike for completely hosted solutions such as AWS or GCP there
+is no single OpenStack.
+
+Some deployments are public clouds akin to AWS while others are only
+available within a single organisation, possibly disconnected from the
+internet. Moreover, the actual services and configuration can vary
+greatly between deployments.
+
+The OpenStack UPI should be able to support scenarios such as these:
+
+* Lack of Swift Object Storage (which we use for serving the bootstrap Ignition config as well as the registry storage)
+* Lack of floating IP addresses (which we use for the public API and ingress endpoints as well as for bootstrap log gathering via must-gather)
+* Disconnected from the internet
```

Comment:
Does this include Provider Network clusters?

Reply:
The middle point (lack of floating IPs) is about provider networks. I wanted to be more descriptive

Reply:
I've updated the wording. Hopefully it's clearer now.